components/brave_vpn/browser/brave_vpn_service.cc

```diff
@@ -506,6 +511,11 @@ void BraveVpnService::RequestCredentialSummary(const std::string& domain) {
 
 void BraveVpnService::OnCredentialSummary(const std::string& domain,
                                           const std::string& summary_string) {
+  if (!skus::DomainIsForProduct(domain, "vpn")) {
```

Comment:
Q) We could hit here if we early return for `vpn` domain from `LoadPurchasedState()`?

Reply:
It shouldn't be possible - but I thought it was worth checking (in case we get bad data from server).